compute_tools/src/extension_server.rs

```diff
@@ -0,0 +1,244 @@
+// Download extension files from the extension store
+// and put them in the right place in the postgres directory (share / lib)
+/*
+The layout of the S3 bucket is as follows:
+├── 111
+    │   ├── v14
+    │   │   ├── extensions
+    │   │   │   ├── anon.tar.zst
+    │   │   │   └── embedding.tar.zst
+    │   │   └── ext_index.json
+    │   └── v15
+    │       ├── extensions
+    │       │   ├── anon.tar.zst
+    │       │   └── embedding.tar.zst
+    │       └── ext_index.json
+    ├── 112
+    │   ├── v14
+    │   │   ├── extensions
+    │   │   │   └── anon.tar.zst
+    │   │   └── ext_index.json
+    │   └── v15
+    │       ├── extensions
+    │       │   └── anon.tar.zst
+    │       └── ext_index.json
+    └── 113
+        ├── v14
+        │   ├── extensions
+        │   │   └── embedding.tar.zst
+        │   └── ext_index.json
+        └── v15
+            ├── extensions
+            │   └── embedding.tar.zst
+            └── ext_index.json
+
+Note that build number cannot be part of prefix because we might need extensions
+from other build numbers.
+
+ext_index.json stores the control files and location of extension archives
+
+We do not duplicate extension.tar.zst files.
+We only upload a new one if it is updated.
+*access* is controlled by spec
```

Comment:
How exactly it is controlled in this layout?
I don't see anything that differs custom and public extensions.

Reply:
there is no distinction between custom and public extensions except for who can access them. 

the idea is `spec` passes a list of the extensions available to the tenant
which will include public extensions and any custom extensions enabled for that tenant

currently this is hack-enabled in `endpoint.rs`